### ISSUE TITLE
How to use GORBID for books?

### ISSUE THREAD

**Issue by keto33** [kermitt2/grobid#673]
GORBID has been designed for journal articles, but I wonder if there is an approach to optimise it for books?

The first issue is the token limit, which can be increased in the settings, but I think there is a reason for the default limit of 1000000 tokens.

_My questions are:_

1. Is it safely to massively increase the token limit to fit the book size?

2. Is there any module or method for dividing a book into its chapters before parsing, since a book chapter is very similar to a journal article (particularly edited books)?

**Comment by kermitt2** [comment]
Hello @keto33 !

Grobid does not support book, there is no way actually to optimize it to process books for the moment. 

1. The 1000000 limit tokens is to prevent JVM out of memory in scenario where we process millions of PDF - we don't want the server to crash with weird huge PDF. There is no particular problem to increase this number, however it does help for book because GROBID is for the moment unaware of the global structure of a book. 

2. There's a branch adding a model to structure "book" level (segmentation into chapters, table of content, table of figures, index, etc.), to support conference proceedings, thesis, books, etc. but it is only early development and not usable. 
https://github.com/kermitt2/grobid/tree/monographe-opaline
It would require CC-BY training data in particular.

**Comment by berserkwarwolf** [comment]
Any progress on this ? @kermitt2 Patrice, can we talk about how can I help to fast track this ?

**Comment by keto33** [comment]
One possible approach is to add the option of page number. I mean parsing a PDF file from page x to page y (instead of parsing the entire file). Then, we can select a chapter of a book starting from page x to page y. This is very beneficial for edited books, as each chapter is very similar to a journal article; having authors, affiliations, references, etc.

**Comment by kermitt2** [comment]
This page slice is already available, see: 
https://github.com/kermitt2/grobid/blob/master/grobid-service/src/main/java/org/grobid/service/GrobidRestService.java#L207

however one can simply slice the PDF before sending it to Grobid (avoid transferring a PDF larger than required and associated slow parsing time). I guess the outline element embedded in the PDF could help for some well generated proceedings.

**Comment by keto33** [comment]
The slicing feature works perfectly! It's amazing that you've already thought of it.

In my experience, it is much faster to use the clicking feature of grobid than slicing the PDF file beforehand. Saving a slice of a PDF file is slow, as all used dictionaries and resources should be copied.

As I checked, it is much faster to parse a book's chapters by start and end pages than splitting the PDF file and feed the chapters as separate inputs to grobid.

**Comment by kermitt2** [comment]
> As I checked, it is much faster to parse a book's chapters by start and end pages than splitting the PDF file and feed the chapters as separate inputs to grobid.

Thank you for the feedback, good to know ! So I should probably document the `start` and `end` parameter in the web service API, this use case is good a reason.

**Comment by keto33** [comment]
Yes, they are very handy and should be documented. Their usage is not limited to books. Many article files have unnecessary pages (publisher cover, appendix, supplementary materials, etc).

Please mention that they are optional (not both of them are necessary). For example, `--form start=2` will parse from page 2 to the end of the document (knowing the end page is not required).

**Comment by rishabhjoshi** [comment]
I was trying to extract the index present in some books using this tool and I tried extracting text from a few pages of content and index only (split the pdf manually). It seems to capture only a handful of index words for some reason. Tried it on multiple books / pdfs but couldn't get all the index any time. Any suggestions?
